vpr/src/place/noc_place_utils.cpp

```diff
@@ -0,0 +1,643 @@
+    const std::vector<NocTrafficFlowId>* assoc_traffic_flows = noc_traffic_flows_storage.get_traffic_flows_associated_to_router_block(moved_block_router_id);
+
+    // now check if there are any associated traffic flows
+    if (assoc_traffic_flows != nullptr) {
+        // There are traffic flows associated to the current router block so process them
+        for (auto& traffic_flow_id : *assoc_traffic_flows) {
+            // first check to see whether we have already re-routed the current traffic flow and only re-route it if we haven't already.
+            if (updated_traffic_flows.find(traffic_flow_id) == updated_traffic_flows.end()) {
+                // now update the current traffic flow by re-routing it based on the new locations of its src and destination routers
+                re_route_traffic_flow(traffic_flow_id, noc_traffic_flows_storage, noc_model, noc_flows_router, placed_cluster_block_locations);
+
+                // now make sure we don't update this traffic flow a second time by adding it to the group of updated traffic flows
+                updated_traffic_flows.insert(traffic_flow_id);
+
+                // update global datastructures to indicate that the current traffic flow was affected due to router cluster blocks being swapped
+                affected_traffic_flows[number_of_affected_traffic_flows] = traffic_flow_id;
+                number_of_affected_traffic_flows++;
+            }
+        }
+    }
+
+    return;
+}
+
+void revert_noc_traffic_flow_routes(const t_pl_blocks_to_be_moved& blocks_affected) {
+    // provides the positions where the affected blocks have moved to
+    auto& place_ctx = g_vpr_ctx.placement();
+    auto& noc_ctx = g_vpr_ctx.mutable_noc();
+
+    NocTrafficFlows* noc_traffic_flows_storage = &noc_ctx.noc_traffic_flows_storage;
+
+    // keeps track of traffic flows that have been reverted
+    // This is useful for cases where two moved routers were part of the same traffic flow and prevents us from re-routing the same flow twice.
+    std::unordered_set<NocTrafficFlowId> reverted_traffic_flows;
+
+    // go through the moved blocks and process them only if they are NoC routers
+    for (int iblk = 0; iblk < blocks_affected.num_moved_blocks; ++iblk) {
+        ClusterBlockId blk = blocks_affected.moved_blocks[iblk].block_num;
+
+        // check if the current moved block is a noc router
+        if (noc_traffic_flows_storage->check_if_cluster_block_has_traffic_flows(blk)) {
+            // current block is a router, so re-route all the traffic flows it is a part of //
+
+            // get all the associated traffic flows for the logical router cluster block
+            const std::vector<NocTrafficFlowId>* assoc_traffic_flows = noc_traffic_flows_storage->get_traffic_flows_associated_to_router_block(blk);
+
+            // now check if there are any associated traffic flows
+            if (assoc_traffic_flows != nullptr) {
```

Comment:
Again, I think this could be recoded to use traffic flow ids instead of a pointer.

Reply:
Added to an issue and will be updated later